rooms/dojo.py

```diff
@@ -0,0 +1,345 @@
+        """
+        This function randomly assign room to a person
+        from a list of rooms that are available_room
+        """
+        available_rooms = self.get_available_rooms(room_type)
+        if len(available_rooms) > 0:
+            room = random.choice(available_rooms)
+            if not room.name in self.allocated:
+                self.allocated[room.name] = []
+            self.allocated[room.name].append(person)
+            return room
+        else:
+            if room_type == Office:
+                self.unallocated["office"].append(person)
+            else:
+                self.unallocated["livingspace"].append(person)
+            return None
+
+    def print_room(self, room_name):
+        """"
+        This function prints names of all allocated
+        members of the passed room
+        """
+        print_out = ""
+        if room_name in self.allocated:
+            for person in self.allocated[room_name]:
+                print_out += person.name.upper() + "\n"
+        else:
+            print_out = "No allocation to this room"
+        print_out = room_name.upper() + "\n" + ("-" * 15) + "\n" + print_out
+        print(print_out)
+
+    def print_allocation(self, file_name=None):
+        """
+        This function prints out all allocated rooms
+        and their allocated members
+        """
+        print_out = ""
+        for room in self.allocated:
+            names = ""
+            for person in self.allocated[room]:
+                names += person.name + ", "
+            names = names[:-2]
+            print_out += room + "\n" + ("-" * len(names)) + \
+                         "\n" + names + "\n"
+        if print_out == "":
+            print("Nobody on the allocated list.")
+        else:
+            self.write_to_file(print_out, file_name)
+            print(print_out.upper())
+
+    def print_unallocated(self, file_name=None):
+        """ This function prints the list of unallocated persons"""
+        print_out = ""
+        for key in self.unallocated:
+            for person in self.unallocated[key]:
+                print_out += person.name.upper() + " - NO " + \
+                                            key.upper() + "\n"
+        if print_out == "":
+            print("Nobody on the unallocated list.")
+        else:
+            print_out = "UNALLOCATED LIST\n\n" + print_out
+            self.write_to_file(print_out, file_name)
+            print(print_out.upper())
+
+    def write_to_file(self, print_out, file_name):
+        """ This function write a string to the file_name specified"""
+        if file_name is not None:
+            file = open("data/%s.txt" % file_name, "w" )
+            file.write(print_out.upper())
+            file.close()
+            print("List have been successfully written to file")
+
+    def reset(self):
+        """ This function reset Dojo to it initializatio stage"""
+        self.__init__()
+
+    def check_valid_id(self, input_val):
```

Comment:
Please update this method's docstring.
You could provide a brief description of the algorithm you are using to check the validity of the id supplied.